pkg/federation/sdlmerge/remove_scalar_duplicates.go

```diff
@@ -0,0 +1,52 @@
+package sdlmerge
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+type removeDuplicateScalarTypeDefinitionVisitor struct {
+	operation     *ast.Document
+	scalarSet     map[string]bool
+	nodesToRemove []ast.Node
+	lastRef       int
+}
+
+func newRemoveDuplicateScalarTypeDefinitionVistior() *removeDuplicateScalarTypeDefinitionVisitor {
+	return &removeDuplicateScalarTypeDefinitionVisitor{
+		nil,
+		make(map[string]bool),
+		make([]ast.Node, 0),
+		-1,
+	}
+}
+
+func (r *removeDuplicateScalarTypeDefinitionVisitor) EnterDocument(operation, definition *ast.Document) {
+	r.operation = operation
+}
+
+func (r *removeDuplicateScalarTypeDefinitionVisitor) EnterScalarTypeDefinition(ref int) {
+	if ref <= r.lastRef {
+		return
+	}
+	name := r.operation.ScalarTypeDefinitionNameString(ref)
+	if ok := r.scalarSet[name]; ok {
```

Comment:
why not `if r.scalarSet[name] {` 
:-)

Reply:
Oh, I thought it would error if the key did not exist.